File: frontend/src/types/property.ts
export type PropertyStatus =
  | 'Draft'
  | 'Active'
  | 'Available'
  | 'Sold'
  | 'Rented'
  | 'Expired'
  | 'Hidden'
  | 'draft'
  | 'verified'
  | 'published'
  | 'expired'
  | string;

export type ListingType = 'Sale' | 'Rent' | string;

export interface PropertyImage {
  id: string;
  propertyId?: string;
  url: string;
  fileName: string;
  originalFileName: string;
  filePath?: string;
  contentType?: string;
  size?: number;
  isThumbnail: boolean;
  sortOrder: number;
  createdAt?: string;
}

export interface Property {
  id: string;
  title: string;
  description?: string | null;
  code?: string;
  price: number;
  area: number | string;
  acreage?: number;
  priceUnit?: string | null;
  address: string;
  provinceId?: string | null;
  provinceName?: string | null;
  districtId?: string | null;
  districtName?: string | null;
  wardId?: string | null;
  wardName?: string | null;
  categoryId?: string | null;
  categoryName?: string | null;
  propertyCategoryId?: string | null;
  propertyCategoryName?: string | null;
  typeId?: string | null;
  typeName?: string | null;
  propertyTypeId?: string | null;
  propertyTypeName?: string | null;
  projectId?: string | null;
  status: PropertyStatus;
  listingType?: ListingType;
  bedrooms?: number;
  bathrooms?: number;
  floors?: number;
  direction?: string | null;
  legalStatus?: string | null;
  images?: PropertyImage[];
  imageUrl: string;
  source: string;
  sourceType?: string | null;
  sourceUrl?: string | null;
  slug?: string | null;
  metaTitle?: string | null;
  metaDescription?: string | null;
  aiScore?: number;
  createdAt: string;
  updatedAt?: string | null;
}

export interface PropertyQuery {
  page: number;
  pageSize: number;
  search?: string;
  keyword?: string;
  minPrice?: number | null;
  maxPrice?: number | null;
  minArea?: number | null;
  maxArea?: number | null;
  provinceId?: string | null;
  districtId?: string | null;
  wardId?: string | null;
  categoryId?: string | null;
  typeId?: string | null;
  status?: string | null;
  sortBy?: 'createdAt' | 'price' | 'area' | 'title' | string;
  sortDirection?: 'asc' | 'desc' | string;
}

export interface PropertyPayload {
  title: string;
  description?: string | null;
  code?: string | null;
  price: number;
  area: number;
  priceUnit?: string | null;
  address?: string | null;
  provinceId?: string | null;
  districtId?: string | null;
  wardId?: string | null;
  categoryId?: string | null;
  propertyCategoryId?: string | null;
  typeId?: string | null;
  propertyTypeId?: string | null;
  projectId?: string | null;
  status: string;
  listingType?: string;
  bedrooms?: number | null;
  bathrooms?: number | null;
  floors?: number | null;
  direction?: string | null;
  legalStatus?: string | null;
}

export interface PropertyCategory {
  id: string;
  name: string;
  slug: string;
  description?: string | null;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string | null;
}

export interface PropertyCategoryPayload {
  name: string;
  slug?: string | null;
  description?: string | null;
  isActive: boolean;
}

export interface PropertyType {
  id: string;
  name: string;
  slug?: string | null;
  description?: string | null;
  icon?: string | null;
  sortOrder?: number;
  isActive: boolean;
}

export interface LocationOption {
  id: string;
  name: string;
  code?: string | null;
  level?: number;
  parentId?: string | null;
  isActive: boolean;
}

export interface PropertyFilter {
  keyword?: string;
  search?: string;
  status?: PropertyStatus;
  area?: string | number;
  categoryId?: string | null;
  typeId?: string | null;
  provinceId?: string | null;
  districtId?: string | null;
  wardId?: string | null;
}

File: backend/src/RealSync.Core/Entities/Property.cs
namespace RealSync.Core.Entities;

/// <summary>
/// Bất động sản — entity chính của hệ thống.
/// Schema theo database-guide.md Section 4.1.
/// </summary>
public class Property : BaseEntity
{
    // Basic Info
    public string PropertyCode { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }

    // Type & Project
    public Guid? PropertyCategoryId { get; set; }
    public PropertyCategory? PropertyCategory { get; set; }
    public Guid PropertyTypeId { get; set; }
    public PropertyType PropertyType { get; set; } = null!;
    public Guid? ProjectId { get; set; }
    public Project? Project { get; set; }

    // Location
    public Guid? AreaId { get; set; }
    public Area? Area { get; set; }
    public string? Address { get; set; }
    public string? Ward { get; set; }
    public string? District { get; set; }
    public string? Province { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }

    // Specs
    public decimal Area_ { get; set; }  // Diện tích (m²) — tên tránh conflict với navigation
    public decimal Price { get; set; }
    public string? PriceUnit { get; set; } = "VND";
    public int Bedrooms { get; set; } = 0;
    public int Bathrooms { get; set; } = 0;
    public int Floors { get; set; } = 1;
    public string? Direction { get; set; }
    public string? LegalStatus { get; set; }

    // Status
    public string Status { get; set; } = "Draft";  // Draft, Active, Sold, Rented, Expired
    public string ListingType { get; set; } = "Sale";  // Sale, Rent
    public int FeaturedLevel { get; set; } = 0;  // 0=Normal, 1=Featured, 2=VIP

    // Source
    public string? SourceType { get; set; }  // Manual, Crawled, Imported
    public string? SourceUrl { get; set; }
    public Guid? CrawlJobId { get; set; }
    public CrawlJob? CrawlJob { get; set; }

    // SEO
    public string? Slug { get; set; }
    public string? MetaTitle { get; set; }
    public string? MetaDescription { get; set; }

    // Navigation
    public ICollection<PropertyImage> Images { get; set; } = new List<PropertyImage>();
}
